client/components/check-icon.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+export default class CheckIcon extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      restaurantsDbAliases: props.restaurantsDbAliases,
+      restaurant: {
+        alias: props.restaurant.alias,
+        url: props.restaurant.url,
+        imageUrl: props.restaurant.image_url,
+        name: props.restaurant.name,
+        location: props.restaurant.location,
+        rating: props.restaurant.rating,
+        reviewCount: props.restaurant.review_count
+      }
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(event) {
+    if (event.target.className === 'far fa-check-circle unchecked') {
+      const reqPost = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(this.state.restaurant)
+      };
+      fetch('/api/save', reqPost)
+        .then(res => res.json())
+        .then(results => {
+          event.target.className = 'fas fa-check-circle checked';
+        });
+    } else {
+      event.target.className = 'far fa-check-circle unchecked';
+    }
+  }
+
+  render() {
+    const aliasCheck = this.state.restaurant.alias;
+    const { restaurantsDbAliases: aliases } = this.state;
+    const alias = aliases.filter(aliasObject => aliasObject.alias === aliasCheck);
```

Comment:
Instead of calculating this during `render`, do it during `componentDidMount` and then setState for `isSelected`.